### ISSUE TITLE
Support for LB service creation with elaborate params via UI

### ISSUE THREAD

**Issue by alena1108** [rancher/rancher#15070]
Today in Rancher UI there is only way to create an LB service for the workload - by setting up the port of LB type on the workload, so the underlying controller would create the LB service automatically. 

The current limitation is you can't really set any extra parameters for the LB service like cloud provider specific annotations or session affinity. We should add support for it by allowing user to explicitly create a Load Balancer service with all the extra parameters via Services page, in addition to implicit-via-workload-ports way.

**Comment by alena1108** [comment]
Parameters we are interested in for LB:

* Session affinity
* Annotations

**Comment by loganhz** [comment]
@alena1108 @vincent99 
Service Discovery page shows the dnsRecord list. But dnsRecord doesn't have `sessionAffinity` field in schema. Do we need backend change before we can implement this?

**Comment by deniseschannon** [comment]
Without this, we can't create internal LB in AKS.

**Comment by billabongrob** [comment]
+1 ^ That's our exact issue.  It can be done via imported YAML, but not in UI.  In an ideal world, the UI would recognize the cluster cloud provider and allow an internal/external LB selection that would automatically apply the appropriate annotations.

**Comment by loganhz** [comment]
@vincent99 @alena1108 any idea?

> Service Discovery page shows the dnsRecord list. But dnsRecord doesn't have sessionAffinity field in schema. Do we need backend change before we can implement this?

**Comment by vincent99** [comment]
Yes? If the field is missing then it needs to be in the API for the UI to support it.

**Comment by loganhz** [comment]
@vincent99 It looks like we already have all fields which kubernetes API has in `/schemas/service` 

Do you think we can abandon `/schemas/dnsRecord` and just use `/schemas/service` for `Service Discovery` tab?

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#service-v1-core

**Comment by davidnuzik** [comment]
Version: master 2.2 (1/15/19)
Confirmed enhancements. I was successfully able to set up a LB service (L4 LB with AKS) on my test nginx workload. It successfully routed per the port mapping and the front end UI was validated (HTTP w/ create, edit, delete, etc) and API as well as on the backend we ensured that our settings correctly ended up on the cluster (check json output for our lb service). Halo tested.
